packages/storybook-builder/src/esbuild-plugin-commonjs-named-exports.ts

```diff
@@ -1,35 +1,76 @@
 import type { Plugin } from 'esbuild';
+import { readFile } from 'fs-extra';
+import { dirname } from 'path';
 
-export function esbuildPluginCommonjsNamedExports(module: string, namedExports: string[]): Plugin {
+export function esbuildPluginCommonjsNamedExports(modules: string[]): Plugin {
   return {
     name: 'commonjs-named-exports',
-    setup(build) {
-      build.onResolve({ filter: new RegExp(`^${module}$`) }, args => {
+    async setup(build) {
+      const { init, parse } = await import('cjs-module-lexer');
+      await init();
+
+      build.onResolve({ filter: new RegExp(`^(${modules.join('|')})$`) }, async args => {
+        if (args.pluginData?.preventInfiniteRecursion) return;
+
+        const { path, ...rest } = args;
+        rest.pluginData = { preventInfiniteRecursion: true };
+        const resolveResult = await build.resolve(path, rest);
+        const resolvePath = resolveResult.path;
+
+        // skip if resolved to an ESM file
+        if (resolvePath.endsWith('.mjs')) return;
```

Comment:
what if it's a project with type module?

Reply:
it's OK, the lexer would try to parse them and if it works, then it will give the reexports